plsql-frontend/src/main/java/org/sonar/plugins/plsqlopen/api/PlSqlGrammar.java

```diff
@@ -442,8 +442,8 @@ private static void createStatements(LexerfulGrammarBuilder b) {
         
         b.rule(CASE_STATEMENT).is(
                 b.optional(LABEL), 
-                CASE, b.optional(OBJECT_REFERENCE),
```

Comment:
I didn't understand why you changed this to a firstOf call. If you keep the old code here and just replace this OBJECT_REFERENCE by EXPRESSION your tests still pass. Is there any reason to not do this?